api/src/main/resources/org/openmrs/module/isanteplusreports/sql/healthQualReports/pediatricReceivedPcrTest.sql

```diff
@@ -42,16 +42,36 @@ WHERE
         LEFT JOIN isanteplus.patient_prescription pp
             ON phv.patient_id = pp.patient_id
         WHERE
-            DATE(phv.visit_date) BETWEEN :startDate AND :endDate
-            OR (
-                DATE(pp.visit_date) BETWEEN :startDate AND :endDate
-                AND pp.rx_or_prophy = 138405
+            (
+	            DATE(phv.visit_date) BETWEEN :startDate AND :endDate AND phv.encounter_type IN (9,10) -- Paeds initial and followup encounter types
+	            OR 
+	            DATE(pp.visit_date) BETWEEN :startDate AND :endDate
             )
+		    AND TIMESTAMPDIFF(MONTH, p.birthdate, phv.visit_date) <= 12
+		    AND TIMESTAMPDIFF(WEEK, p.birthdate, phv.visit_date) >= 4
+	)
+	
+	AND p.patient_id IN ( -- Eligible for PCR
+		SELECT DISTINCT patient_id -- Exposed to HIV between 4 weeks and 12 months
+        FROM isanteplus.pediatric_hiv_visit v
+        WHERE 
+            actual_vih_status = 1405 ANd v.encounter_date <= :endDate
+            AND TIMESTAMPDIFF(WEEK, p.birthdate, v.encounter_date) >= 4
+            AND TIMESTAMPDIFF(MONTH, p.birthdate, v.encounter_date) <= 12
+		UNION		
+        SELECT DISTINCT pvt.patient_id -- Exposed to HIV between 12 and 18 months with a +ve HIV antigen test
+        FROM isanteplus.pediatric_hiv_visit v
+        INNER JOIN isanteplus.virological_tests pvt ON v.patient_id = pvt.patient_id
+        WHERE 
+            pvt.answer_concept_id = 1030
```

Comment:
Most of these hard coded values, will pose a challenge in future. Maybe you need to create a file that contains all the resoucres incluing cocnepts, encounters etc and just reference by name. When there will be a need to change that concepts to something else, it would be easier to change in one place

Reply:
I agree... However, my main aim here was to ensure that the data being pulled is accurate and that the code reflects the definition in the indicator definition document.

For subsequent indicator reports, I will incorporate these suggestions. I will also plan to refactor this code later once all the other pull requests relating to the Health Qual Report have been merged.